### DISCUSSION TITLE
[@types/react] Argument of type 'Element' is not assignable to parameter of type 'ReactNode' with v18.2.22

### DISCUSSION THREAD

**Discussion by stephent** [DefinitelyTyped/DefinitelyTyped#66754] [Issues with a @types package] [answered]
I'm seeing an issue after updating to @types/react 18.2.22. In a standard Vite React app, in App.tsx, I have;

```
import React from 'react';
import ReactDOM from 'react-dom/client';

import App from './App.tsx';
import './index.css';

ReactDOM.createRoot(document.getElementById('root')!).render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
);
```

When I try to build the project, by running `tsc`, it fails with the following error:

```
src/main.tsx:8:3 - error TS2345: Argument of type 'Element' is not assignable to parameter of type 'ReactNode'.
  Property 'children' is missing in type 'Element' but required in type 'ReactPortal'.
```

When I revert to 18.2.21, the issue is gone. Anyone else seeing this problem?

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment]
Do you have multiple versions of `@types/react` installed? Could you read and follow https://solverfox.dev/writing/duplicate-types-react and let me know if this doesn't fix your issue?

**Reply by stephent** [reply, marked as answer]
@eps1lon wouldn't you know, after updating once again to 18.2.22, all is well - the above error no longer appears on building. On checking for installed versions, everything seems consistent:

```
npm ls @types/react
...
├─┬ @react-three/fiber@8.14.2
│ ├─┬ @types/react-reconciler@0.26.7
│ │ └── @types/react@18.2.22 deduped
│ └─┬ its-fine@1.1.1
│   └─┬ @types/react-reconciler@0.28.4
│     └── @types/react@18.2.22 deduped
├─┬ @types/react-dom@18.2.7
│ └── @types/react@18.2.22 deduped
├── @types/react@18.2.22
└─┬ jotai@2.4.2
  └── @types/react@18.2.22 deduped
```

Since I encountered the original error, I switched the project back to `npm` instead of `yarn` for unrelated reasons. Maybe that cleaned something up en route.

Thanks for your help!

**Comment by magicdawn** [comment]
after a [npm-check-updates](https://www.npmjs.com/package/npm-check-updates) action, pnpm gives such structure

`pnpm why @types/react`

```
devDependencies:
@types/react 18.2.22
@types/react-dom 18.2.7
└── @types/react 18.2.21
```

and in pnpm-lock.yaml

```
/@types/react-dom@18.2.7:
    resolution: {integrity: sha512-xxx}
    dependencies:
      '@types/react': 18.2.21
```

- delete node_modules, and `pnpm install` will not fix this. 
- in order to update pnpm-lock.yaml
  - downgrade `@types/react-dom`
  - `pnpm install`
  - upgrade `@types/react-dom`
  - `pnpm install`

**Comment by durchanek** [comment]
On yarn, running `yarn dedupe` fixed the issue for me.

**Reply by BenjaminHoch-GoTo** [reply]
yarn dedupe doesn't work for me:
error The dedupe command isn't necessary. `yarn install` will already dedupe.

i had to add 
`  "resolutions": {
    "@types/react": "18.2.22"
  },`

to my package.json, which is not ideal, because it will be stuck to this version, until removed from here

**Reply by BenjaminHoch-GoTo** [reply]
yarn de-duplicate worked fine though :)
